lisa/node.py

```diff
@@ -160,24 +154,49 @@ def check_sudo_password_required(self) -> None:
                 # So check twice to get two kinds of prompt
                 password_prompts = []
                 for i in range(1, 3):
-                    process = self._execute(
-                        f"echo {constants.LISA_TEST_FOR_SUDO}",
-                        shell=True,
-                        sudo=True,
-                        no_info_log=True,
-                    )
-                    result = process.wait_result(10)
+                    result = self._execute_sudo_test()
                     if result.exit_code != 0:
-                        raise RequireUserPasswordException(
-                            "The password might be invalid for running sudo command"
-                        )
+                        # The password might be invalid for running sudo command
+                        # Try to reset password and check again
+                        is_successfull = self._is_password_reset_successful()
+                        if is_successfull:
+                            result = self._execute_sudo_test()
+                        if not is_successfull or result.exit_code != 0:
+                            raise RequireUserPasswordException(
+                                "The password might be invalid for running sudo command"
+                            )
                     password_prompt = result.stdout.replace(
                         f"{constants.LISA_TEST_FOR_SUDO}", ""
                     )
                     password_prompts.append(password_prompt)
                     self.log.debug(f"password prompt {i}: {password_prompt}")
                 ssh_shell.password_prompts = password_prompts
 
+    def _execute_sudo_test(self) -> ExecutableResult:
```

Comment:
1. Return bool value to indicate success or not.
2. Shorter name like `test_sudo`